src/components/ActivityCard/ActivityCard.tsx

```diff
@@ -0,0 +1,274 @@
+    }>;
+    tripId: string;
+}
+
+export default function ActivityCard(props: ActivityCardProps) {
+    const { getCurrentUser } = useCurrentAuthUser();
+    const [userLiked, setUserLiked] = useState(false);
+    const [userDisliked, setUserDisliked] = useState(false);
+    const [userId, setUserId] = useState("");
+    const [executeCreate] = useCreateActivityReactionMutation();
+    const [executeRemove] = useRemoveActivityReactionMutation();
+
+    const date: Date = new Date(Date.parse(props.date || ""));
+
+    getCurrentUser().then((user) => {
+        if (user == null) {
+            return;
+        }
+        setUserId(user.id);
+
+        setUserLiked(
+            props.activityReactions.filter(
+                (r) =>
+                    r.activityReactionType ===
+                        Types.ActivityReactionType.Like &&
+                    r.addedByUser.id === user.id
+            ).length > 0
+        );
+
+        setUserDisliked(
+            props.activityReactions.filter(
+                (r) =>
+                    r.activityReactionType ===
+                        Types.ActivityReactionType.Dislike &&
+                    r.addedByUser.id === user.id
+            ).length > 0
+        );
+    });
+
+    const handleReactionType = (reactionType: Types.ActivityReactionType) => {
+        const remove = () => {
+            executeRemove({
+                variables: {
+                    input:
+                        props.activityReactions.find(
+                            (r) => r.addedByUser.id === userId
+                        )?.id || "",
+                },
+                refetchQueries: [
+                    refetchGetTripQuery({
+                        tripId: props.tripId,
+                    }),
+                ],
+            }).catch((e) => console.log(e)); // TODO error handling
+        };
+
+        const create = (activityReactionType: Types.ActivityReactionType) => {
+            executeCreate({
+                variables: {
+                    input: {
+                        activityId: props.id,
+                        activityReactionType: activityReactionType,
+                    },
+                },
+                refetchQueries: [
+                    refetchGetTripQuery({
+                        tripId: props.tripId,
+                    }),
+                ],
+            }).catch((e) => console.log(e)); // TODO error handling
+        };
+        if (userLiked) {
+            // check if user has already liked activity
+            remove();
+            if (reactionType === Types.ActivityReactionType.Dislike) {
+                create(Types.ActivityReactionType.Dislike);
+            }
+        } else if (userDisliked) {
+            // check if user has already disliked activity
+            remove();
+            if (reactionType === Types.ActivityReactionType.Like) {
+                create(Types.ActivityReactionType.Like);
+            }
```

Comment:
I think we should use update for changing the type and remove when removing the reaction.

```suggestion
            if (reactionType === Types.ActivityReactionType.Like) {
                create(Types.ActivityReactionType.Like);
            } else {
                remove();
            }
```